File: Typescript/test/validStrings.test.ts
import { expect, test } from '@jest/globals';
import * as solution from '../src/validStrings';

test('Example 1', () => {
	let input = 3;
	let expected = ["010", "011", "101", "110", "111"];
	expect(solution.validStrings(input)).toEqual(expected);
});

test('Example 2', () => {
	let input = 1;
	let expected = ["0", "1"];
	expect(solution.validStrings(input)).toEqual(expected);
});

test('Generated strings never contain adjacent zeros', () => {
	let input = 4;
	let actual = solution.validStrings(input);
	expect(actual.every((value) => !value.includes("00"))).toBe(true);
});

File: Dart/test/validStrings_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:leetcode/validStrings.dart';

void main() {
  test("test example 1", () {
    var input = 3;
    var expected = ["010", "011", "101", "110", "111"];

    final sln = Solution();
    var output = sln.validStrings(input);
    expect(output, expected);
  });
  test("test example 2", () {
    var input = 1;
    var expected = ["0", "1"];

    final sln = Solution();
    var output = sln.validStrings(input);
    expect(output, expected);
  });
}
